src/main/java/io/netty/incubator/codec/http3/QpackDecoder.java

```diff
@@ -0,0 +1,250 @@
+/*
+ * Copyright 2020 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.incubator.codec.http3;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.util.AsciiString;
+
+import static io.netty.util.internal.ObjectUtil.checkPositive;
+import static io.netty.incubator.codec.http3.Http3Headers.PseudoHeaderName.getPseudoHeader;
+import static io.netty.incubator.codec.http3.Http3Headers.PseudoHeaderName.hasPseudoHeaderFormat;
+
+final class QpackDecoder {
+
+    private static final Http3Exception HEADER_ILLEGAL_INDEX_VALUE =
+        new Http3Exception("QPACK - illegal index value");
+    private static final Http3Exception NAME_ILLEGAL_INDEX_VALUE =
+        new Http3Exception("QPACK - illegal index value");
+
+    private static final long DEFAULT_MAX_HEADER_LIST_SIZE = 0xffffffffL;
+
+    private final QpackHuffmanDecoder huffmanDecoder = new QpackHuffmanDecoder();
```

Comment:
Just to clarify... This could be just the "standard" HuffmanDecoder that would be shared with http2 in the future, correct @kachayev ?

Reply:
Yes, exactly. The only thing we will need to change: https://github.com/netty/netty-incubator-codec-http3/blob/8110eccd9637f8c2836e7b1ab0d24d9a8bb15bbd/src/main/java/io/netty/incubator/codec/http3/QpackHuffmanDecoder.java#L4649

Reply:
When moving to `codec` package, this exception should be changed to something fairly generic. Meaning that the code should wrap decoding into try/catch to produce meaningful error information "locally"